orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java

```diff
@@ -24,18 +28,46 @@ public class OrchestrationChatResponse {
    */
   @Nonnull
   public String getContent() throws OrchestrationClientException {
-    final var choices =
-        ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult()).getChoices();
-
-    if (choices.isEmpty()) {
-      return "";
-    }
-
-    final var choice = choices.get(0);
+    final var choice = getCurrentChoice();
 
     if ("content_filter".equals(choice.getFinishReason())) {
       throw new OrchestrationClientException("Content filter filtered the output.");
     }
     return choice.getMessage().getContent();
   }
+
+  /**
+   * Get the token usage.
+   *
+   * @return The token usage.
+   */
+  @Nonnull
+  public TokenUsage getTokenUsage() {
+    return ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult()).getUsage();
+  }
+
+  /**
+   * Get all messages. This can be used for subsequent prompts as a message history.
+   *
+   * @return A list of all messages.
+   */
+  @Nonnull
+  public List<ChatMessage> getAllMessages() {
+    final var messages = originalResponse.getModuleResults().getTemplating();
+
+    messages.add(getCurrentChoice().getMessage());
+    return messages;
+  }
+
+  /**
+   * Get list of choices.
+   *
+   * @return A list of choices.
+   */
+  @Nonnull
+  private LLMChoice getCurrentChoice() {
+    return ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult())
+        .getChoices()
+        .get(0);
+  }
```

Comment:
1. Incorrect javadoc
2. what happened to null checks?

Reply:
No null check needed IMO because the server should never return empty choices. If it does, throwing an NPE is fine for that case IMO, since we in general don't validate the server response.

Reply:
As discussed, solved by adding a comment in the code.